### PR TITLE
Allow pasting image from clipboard

### PR DESCRIPTION
**Pull request by mdsaban**
## Description

The editor currently only supports copying and pasting images from the internet. It cannot paste images from the local clipboard, like screenshots. This PR adds support for pasting images directly from the clipboard.

https://github.com/user-attachments/assets/c18f1470-44a4-4125-82ee-3c043d531d09

Related issue #320

## Type of change

Please tick the relevant option.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by mdsaban** [comment]
@Darginec05 I've created this PR as a draft. I have a few questions I need your help with:

What's the best way to create a block for the pasted image? Currently, I’ve implemented it in a basic manner for the POC.
How can we extend this to allow users to upload the pasted image to their servers if needed?
Should we handle the uploading functionality in a separate PR to keep this one focused, or include it here?

**Comment by Darginec05** [comment]
@mdsaban So, I've checked your PR and there are some problems which I want to mention.
 
First, we cannot predict that the Image plugin will be passed to the editor, as [implemented here](https://github.com/Darginec05/Yoopta-Editor/pull/340/files#diff-a9755db82634c3c3849f41b06584318f69371897dee0671fae55861340ae3416R31)

I think it should be more abstract, at the plugin level. 

In fact, I initially thought that we should implement this at the plugin level in the events.onPaste object, but this will not work if we will into cursor position when cursor is in the Paragraph block.

And we can't just take [the first one items[0]](https://github.com/Darginec05/Yoopta-Editor/pull/340/files#diff-1287f69961082fcc301d76fdabf452ed434009f63907f5ff8f5ed697c14f23deR158) because multiple files can be inserted.

**Comment by Darginec05** [comment]
@mdsaban In general, I think we should handle the onPaste and onDrop events directly in the <Editor /> component. So using event.clipboardData.items from onPaste (event.dataTransfer.items for onDrop) we can understand what the type was dropped into editor

**Comment by Darginec05** [comment]
1. Move event handlers from `SlateEditorComponent` to `<Editor />`
2. Event handlers should be matched with `mimeType` and then get element data via `handler` 
```ts
      onPaste={(event) => {
        const items = event.clipboardData.items;

        for (let i = 0; i < items.length; i++) {
          const item = items[i];

          const file = item.getAsFile();
        }
      }}
      onDrop={(event) => {
        event.preventDefault();

        const items = event.dataTransfer.items;

        for (let i = 0; i < items.length; i++) {
          const item = items[i];

          // match mime type
          if (matchedPluginWithMimeType.clipboardRules.mimeTypes.includes(item.type)) {
            const file = item.getAsFile();

            const element = matchedPluginWithMimeType.clipboardRules.handler(file);
            const blockData = Blocks.buildBlockData({ type: matchedPluginWithMimeType.type, value: [element] });
            Blocks.insertBlock(editor, matchedPluginWithMimeType.type, { blockData: blockData });
          }
        }
      }}```

3. And we need add clipboard rules in Image, Video plugins: 
```ts
const Image = new YooptaPlugin<ImageElementMap, ImagePluginOptions>({
  type: 'Image',
  clipboardRules: {
    mimeTypes: ['image/png', 'image/jpeg', 'image/gif', 'image/webp'],
    handler: (file: File) => {
      const data = Image.getPlugin.options?.onUpload(file);
      return {
        src: data.secure_url,
        sizes: {
          width: data.width,
          height: data.height,
        },
      };
    },
  },
// ...other code
}
```

File: packages/core/editor/src/plugins/types.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ export type Plugin<TElementMap extends Record<string, SlateElement>, TPluginOpti
   events?: PluginEvents;
   options?: PluginOptions<TPluginOptions>;
   parsers?: Partial<Record<PluginParserTypes, PluginParsers>>;
+  clipboardPasteOrDropRules?: Record<string, any>;
```
Comment (mdsaban) [inline packages/core/editor/src/plugins/types.ts:92]: TODO: use correct types for this

**Comment by Darginec05** [comment]
related to https://github.com/yoopta-editor/Yoopta-Editor/issues/369

**Pull request closed without merging**